librosa/core/pitch.py

```diff
@@ -597,19 +584,18 @@ def yin(
 
     # Calculate minimum and maximum periods
     min_period = int(np.floor(sr / fmax))
-    max_period = min(int(np.ceil(sr / fmin)), frame_length - win_length - 1)
+    max_period = min(int(np.ceil(sr / fmin)), frame_length - 1)
 
     # Calculate cumulative mean normalized difference function.
     yin_frames = _cumulative_mean_normalized_difference(
-        y_frames, frame_length, win_length, min_period, max_period
+        y_frames, min_period, max_period
     )
 
     # Parabolic interpolation.
     parabolic_shifts = _parabolic_interpolation(yin_frames)
 
     # Find local minima.
     is_trough = util.localmin(yin_frames, axis=-2)
-    is_trough[..., 0, :] = yin_frames[..., 0, :] < yin_frames[..., 1, :]
```

Comment:
I think this line is still needed; it's here to ensure that the localmin calculation is properly defined at the edge of the array.

Reply:
I suppose it's a question of definition whether rigorously speaking, an edge point should be considered a local minimum or not. However, I don't think it makes a practical difference in this case. The algorithm will only pick lag candidates within the search bounds, and a lag of 0 can't be within those bounds since it would correspond to "infinite frequency". 

The `localmin` call returns a boolean array, where `is_trough[..., 0, :] == False` by definition according to the [`localmin` documentation](https://librosa.org/doc/main/generated/librosa.util.localmin.html) - I think this suits the algorithm well and doesn't require that additional check.

But I don't think the extra check impacts performance or accuracy, so I'd be happy to add it back in.

Reply:
> and a lag of 0 can't be within those bounds since it would correspond to "infinite frequency".

That's true, but `yin_frames` is already trimmed to the min / max period before returning from CNMD, so "0" here actually means min_period.

Reply:
Ah my bad, you're completely right of course! I'll revert this change.